scene/2d/tile_map_layer.cpp

```diff
@@ -2618,14 +2643,23 @@ Vector2i TileMapLayer::get_coords_for_body_rid(RID p_physics_body) const {
 	return bodies_coords[p_physics_body];
 }
 
+Ref<TileSet> TileMapLayer::get_effective_tile_set() const {
+	TileMapLayerGroup *tile_map_layer_group = Object::cast_to<TileMapLayerGroup>(get_parent());
+	if (tile_map_layer_group) {
+		return tile_map_layer_group->get_tileset();
+	} else {
+		return Ref<TileSet>();
+	}
+}
```

Comment:
Note there's still `TileMapLayer::_fetch_tileset` which does the same but assumming `TileMap` instead of `TileMapGroup`. Probably should be removed and all its usages replaced by this? :thinking:

Reply:
Ah right, good idea.